### ISSUE TITLE
Remove [contracts.precondition.override.invalid] warnings while Overriding method of Interfaces

### ISSUE THREAD

**Issue by the1derer** [typetools/checker-framework#2477]
Consider the following code:
```java
import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
import org.checkerframework.checker.nullness.qual.RequiresNonNull;
import org.checkerframework.checker.nullness.qual.EnsuresNonNull;

interface TestInterface {
    public String getField1();
}

class A implements TestInterface {
    private @MonotonicNonNull String field1;

    @EnsuresNonNull("field1")
    public void setField1(String arg) {
        this.field1 = arg;
    }

    @RequiresNonNull("field1")
    public String getField1() {
        return field1;
    }
}
```
Command:
``javacheck -processor nullness FILENAME.java``

This generated the following errors:
```
Test.java:19: error: [contracts.precondition.override.invalid] Precondition must be no stronger than in superclass.
    public String getField1() {
                  ^
  Method
    @Initialized @NonNull String getField1(@Initialized @NonNull A this) in A
    cannot override
    @Initialized @NonNull String getField1(@Initialized @NonNull TestInterface this) in TestInterface
  Precondition is too strong: '@NonNull' for 'this.field1'
1 error
```
As there is no way to annotate methods of Interface with ``@RequiresNonNull``, So this type of situation should be exempted from ``[contracts.precondition.override.invalid]``

**Comment by mernst** [comment]
The warning you showed is demonstrating a flaw in the design of `A`.  We would not want to disable such warnings.

The design flaw is that `A` is not a behavioral subtype of `TestInterface`, even though `A` is a Java subtype of `TestInterface`.
Behavioral subtyping states that a subtype must not impose stronger constraints than a supertype.
`TestInterface` states that `getField1` always returns a non-null value.
`A` does not satisfy this contract.

It would be unsound to permit violation of behavioral subtyping.  Clients would never have any guarantee that their code would not suffer an exception, even if the Checker Framework issued no warnings.
